### ISSUE TITLE
As an API User I want to be able to receive errors when my query is not correct

### ISSUE THREAD

**Issue by dimus** [GlobalNamesArchitecture/gnindex#75]
If my query is wrong I want a response with http code **400 (bad request)** and JSON in the format of

```json
{ "errors": [ {"message": "something is wrong --- error 1", 
               "locations": [{ "line": 1, "column": 3}] }, 
              {"message": "something is wrong --- error 2"} ] }
```

**Comment by dimus** [comment]
From https://www.apollographql.com/docs/react/features/error-handling.html

```
 if (graphQLErrors)
    graphQLErrors.map(({ message, locations, path }) =>
      console.log(
        `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`,
      ),
    );
```

**Comment by dimus** [comment]
Also: 

https://github.com/machinebox/graphql/blob/3a92531802258604bd12793465c2e28bc4b2fc85/graphql.go#L252

and

https://github.com/machinebox/graphql/blob/3a92531802258604bd12793465c2e28bc4b2fc85/graphql_json_test.go#L78

**Comment by dimus** [comment]
From https://graphql.org/learn/validation/

```
{
  "errors": [
    {
      "message": "Cannot spread fragment \"NameAndAppearancesAndFriends\" within itself.",
      "locations": [
        {
          "line": 11,
          "column": 5
        }
      ]
    }
  ]
}
```

**Comment by alexander-myltsev** [comment]
I don't see locations in Sangria structs. It is embedded in the error message string. Do you need locations?

**Comment by dimus** [comment]
Lets skip locations, probably Sangria puts them into error string, so they will be visible.

**Comment by alexander-myltsev** [comment]
Fixed by 7ae67417cf33bb9a2abf865d1b1e74cb67543df5
